zipkin-recorder.go

```diff
@@ -117,10 +117,9 @@ func (r *Recorder) RecordSpan(sp RawSpan) {
 	}
 
 	for _, spLog := range sp.Logs {
-		if spLog.Timestamp.IsZero() {
-			spLog.Timestamp = time.Now()
-		}
-		annotate(span, spLog.Timestamp, spLog.Event, r.endpoint)
+		// XXX: decide how to represent spLog.Fields as an annotation
```

Comment:
heh probably not the best one to comment, as I've never been a fan of this feature. it is a complex datastructure (map) with a timestamp. IMHO, this sort of thing is a front-end rendering or layout problem, in the same space as things like log4j, unless we are defining this as a backend/wire format concern.

Before I speculate how this might be used in zipkin, it is probably best to have the champion of this feature comment. @yurishkuro are you planning to make an encoder on the frontend? or are you developing a non-zipkin backend data structure to replace zipkin? or do you have other plans wrt this thing?

Until then, here's a hint.. Let's assume we didn't assume responsibilities like normal loggers have with regards to formatting, and we just did json encoding.. ex `{"event": "soft error", "type": "cache timeout", "waited.millis": 1500}`

With no changes to the existing UI or data format, it ends up looking like this (the mis-rendering is probably a bug):
![screen shot 2016-09-20 at 9 58 07 am](https://cloud.githubusercontent.com/assets/64215/18655143/fc27429c-7f18-11e6-98bf-c189531cdfd8.png)

And here's it in the [up-and-coming UI](https://github.com/openzipkin/zipkin-ui) by @rogeralsing
![screen shot 2016-09-20 at 9 58 26 am](https://cloud.githubusercontent.com/assets/64215/18655156/171843f8-7f19-11e6-8a1b-30a6ef7b58e1.png)